pkg/ucp/api/v20231001preview/resourcetype_conversion.go

```diff
@@ -61,8 +74,21 @@ func (dst *ResourceTypeResource) ConvertFrom(src v1.DataModelInterface) error {
 
 	dst.Properties = &ResourceTypeProperties{
 		ProvisioningState: to.Ptr(ProvisioningState(dm.InternalMetadata.AsyncProvisioningState)),
+		Capabilities:      to.SliceOfPtrs(dm.Properties.Capabilities...),
 		DefaultAPIVersion: dm.Properties.DefaultAPIVersion,
 	}
 
 	return nil
 }
+
+func validateCapability(input *string) error {
+	if input == nil {
+		return v1.NewClientErrInvalidRequest("capability cannot be null")
+	}
+
+	if *input == datamodel.CapabilitySupportsRecipes {
+		return nil
+	}
+
+	return v1.NewClientErrInvalidRequest(fmt.Sprintf("capability %q is not recognized. Supported capabilities: %s", *input, datamodel.CapabilitySupportsRecipes))
```

Comment:
This validation will be expanded over time, and potentially made more flexible. For right now we're just validating that one capability we support.